stdlib/public/core/UnicodeScalarProperties.swift

```diff
@@ -743,15 +743,17 @@ extension Unicode.Scalar.Properties {
 
 /// Case mapping properties.
 extension Unicode.Scalar.Properties {
-  fileprivate struct _CaseMapping {
+  internal struct _CaseMapping {
     let rawValue: UInt8
 
     static let uppercase = _CaseMapping(rawValue: 0)
     static let lowercase = _CaseMapping(rawValue: 1)
     static let titlecase = _CaseMapping(rawValue: 2)
   }
 
-  fileprivate func _getMapping(_ mapping: _CaseMapping) -> String {
+  internal func _getMapping(
+    _ mapping: _CaseMapping
+  ) -> (scalar: Unicode.Scalar?, specialMapping: UnsafeBufferPointer<UInt8>?) {
```

Comment:
It would be really nice if this could be an enum. You can either define one, or use the existing internal [`_Either`](https://github.com/apple/swift/blob/main/stdlib/public/core/EitherSequence.swift) type.